app/assets/javascripts/discourse/app/static/wizard/components/fields/radio.gjs

```diff
@@ -0,0 +1,75 @@
+import Component from "@glimmer/component";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action, set } from "@ember/object";
+import { hash } from "@ember/helper";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class Radio extends Component {
+  constructor() {
+    super(...arguments);
+
+    this._setSelected();
+  }
+
+  get field() {
+    return this.args.field;
+  }
+
+  @action
+  selectionChanged(input) {
+    this.field.value = input.target.value;
+    this._resetSelected();
+    this._setSelected();
+  }
+
+  _resetSelected() {
+    for (let choice of this.field.choices) {
+      set(choice, "selected", false);
+    }
+  }
+
+  _setSelected() {
+    for (let choice of this.field.choices) {
+      if (this.field.value === choice.id) {
+        set(choice, "selected", true);
+      }
+    }
+  }
+
+  <template>
+    <div class="wizard-container__radio-choices">
+      {{#each @field.choices as |c|}}
+        <div class="wizard-container__radio-choice {{if c.selected 'selected'}}">
```

Comment:
You can use `concatClass` for this use case, e.g.

https://github.com/discourse/discourse/blob/d11d7158538978f7b9f809c6a5f04fdda48af7a9/app/assets/javascripts/float-kit/addon/components/d-menu.gjs#L56-L62

Reply:
Also I think `selected` should be `-selected` in BEM since it's a modifier

Reply:
I think it's double dash actually. I.e. `.block__elem--mod`. We might have gotten this wrong elsewhere.